### PR TITLE
fix(igxForOf): Fix issue with active element blur when inside shadow …

### LINKED ISSUE #13186: Grid error when there is focused element and you try to scroll inside shadow dom
## Steps to reproduce  

Reproducible only in Chrome under shadow DOM.
1. Add grid and add some cell tempate inside which place a button or input group.
2. Run the sample and click over the added button.
3. Scroll the mause wheel or with the scroll bar.
4. Observe the console.

## Result
image

## Expected result  
No error should be thrown

columnTemplatesIssue

### PR DESCRIPTION
**Pull request by MayaKirova**
…DOM.

Closes #13186   

### Additional information (check all that apply):
 - [x] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/directives/for-of/for_of.directive.ts [hunk 1]
```diff
@@ -1022,15 +1022,16 @@ export class IgxForOfDirective<T> implements OnInit, OnChanges, DoCheck, OnDestr
      * Clears focus inside the virtualized container on small scroll swaps.
      */
     protected scrollFocus(node?: HTMLElement): void {
-        const activeElement = this.document.activeElement as HTMLElement;
+        const document = (node.getRootNode() as any);
```
Comment (pmoleri) [inline projects/igniteui-angular/src/lib/directives/for-of/for_of.directive.ts:1025]: `as Document | ShadowRoot`

File: projects/igniteui-angular/src/lib/directives/for-of/for_of.directive.ts [hunk 2]
```diff
@@ -1022,15 +1022,16 @@ export class IgxForOfDirective<T> implements OnInit, OnChanges, DoCheck, OnDestr
...
+        const activeElement = document.activeElement as HTMLElement;
```
Comment (pmoleri) [inline projects/igniteui-angular/src/lib/directives/for-of/for_of.directive.ts:1026]: if you use the previous cast you probably won't need this one.

Comment (MayaKirova) [reply projects/igniteui-angular/src/lib/directives/for-of/for_of.directive.ts:1026]: Resolves to `Element` so still needs this cast to have `blur`.

**Review by pmoleri — APPROVED** [review]
Looks good to me (haven't tested it).

**Review by damyanpetev — APPROVED** [review]

**Pull request merged**